sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FilePartition.scala

```diff
@@ -83,6 +83,24 @@ object FilePartition extends Logging {
     partitions.toSeq
   }
 
+  def getFilePartitions(
+      sparkSession: SparkSession,
+      partitionedFiles: Seq[PartitionedFile],
+      maxSplitBytes: Long): Seq[FilePartition] = {
+    val openCostBytes = sparkSession.sessionState.conf.filesOpenCostInBytes
+    val maxPartitionNum = sparkSession.sessionState.conf.filesMaxPartitionNum
+    val partitions = getFilePartitions(partitionedFiles, maxSplitBytes, openCostBytes)
```

Comment:
After this line, we ignore `spark.sql.files.maxPartitionBytes`. Could you add a warning about this side-effect? This means each task will took a longer time and the job can become slower in some cases.